myTrip/photo/views.py

```diff
@@ -1,38 +1,56 @@
 """Module generate view for photo requests."""
 
 import json
+
 from django.views.generic.base import View
 from django.http import JsonResponse, HttpResponse
+
+from checkpoint.models import Checkpoint
+from registration.models import CustomUser
+from trip.models import Trip
 from .models import Photo
 
 
+
 class PhotoView(View):
     """Class that handle HTTP requests."""
 
-    def get(self, request, photo_id):
-        """GET request handler.Get photo by photo_id"""
+    def get(self, request, trip_id=None, checkpoint_id=None, photo_id=None):
+        """GET request handler. Can return photo/photos ordered by given args."""
+        if not photo_id:
+            photos = Photo.filter(trip_id, checkpoint_id)
+            if not photos:
+                return HttpResponse(status=404)
+            data = [photo.to_dict() for photo in photos]
+            return JsonResponse(data, status=200, safe=False)
         photo = Photo.get_by_id(photo_id)
         if not photo:
             return HttpResponse(status=404)
-        photo = photo.to_dict()
-        return JsonResponse(photo, status=200, safe=False)
+        data = photo.to_dict()
+        return JsonResponse(data, status=200)
 
-    def post(self, request):
-        """POST request hangler.Creating a new photo object"""
+    def post(self, request, trip_id=None, checkpoint_id=None):
+        """POST request hangler.Creating a new photo object and return status 201("created")"""
         post_data = json.loads(request.body.decode('utf-8'))
-        photo = Photo.create(**post_data).to_dict()
-        return JsonResponse(photo, status=201)
+        user = CustomUser.get_by_id(post_data["user"])
```

Comment:
After the implementation of AUTH, you will use request.user.id, you do not need to transfer in request.body